src/main/java/org/mule/extension/runtime/ConfigurationInstanceRegistrationCallback.java

```diff
@@ -21,13 +20,13 @@
 {
 
     /**
-     * Notifies the creations of {@code configurationInstance}
-     * by the given {@code configurationInstanceProvider}
+     * Registers the given {@code configurationInstance}
      *
-     * @param configurationInstanceProvider the {@link ConfigurationInstanceProvider} which created the instance
-     * @param configurationInstance         the created instance
-     * @param <C>                           the type of the created instance
+     * @param extension                 The {@link Extension} model for the {@code configurationInstance}
+     * @param configuration             the {@link Configuration} model for the {@code configurationInstance}
+     * @param configurationInstanceName a unique name for the {@code configurationInstance}
+     * @param configurationInstance     an instance that is a realisation of the {@link Configuration} model
+     * @param <C>                       the generic type of the {@code configurationInstance}
      */
-    <C> void registerNewConfigurationInstance(ConfigurationInstanceProvider<C> configurationInstanceProvider,
-                                              C configurationInstance);
+    <C> void registerConfigurationInstance(Extension extension, Configuration configuration, String configurationInstanceName, C configurationInstance);
```

Comment:
Why do you need the configuration as a parameter? Is not enough with the configuration name?